pkg/cloudprovider/cloudprovider.go

```diff
@@ -193,7 +193,7 @@ func (c *CloudProvider) IsMachineDrifted(ctx context.Context, machine *v1alpha5.
 	}
 	amiDrifted, err := c.isAMIDrifted(ctx, machine, provisioner, nodeTemplate)
 	if err != nil {
-		return false, err
+		return false, cloudprovider.IgnoreMachineNotFoundError(fmt.Errorf("calculating ami drift, %w", err))
```

Comment:
Does this work? This will ways be an error, since you're using fmt.Errorf.

Reply:
Yes it does, wrapping works!